helm/values.yaml

```diff
@@ -1020,151 +992,68 @@ readinessProbe:
+  ## - pg_hba.conf: clients authentication file
+  ## - pg_ident.conf: map external users to internal users
+  ## See: https://cloudnative-pg.io/documentation/1.20/postgresql_conf/
+  ##
+  postgresql:
+    ## @param cnpg.postgresql.pg_hba [default: see values.yaml] client authentication pg_hba.conf
+    ## ref: https://cloudnative-pg.io/documentation/1.20/postgresql_conf/#the-pg_hba-section and
+    ## https://www.postgresql.org/docs/current/static/auth-pg-hba-conf.html
     ## We override the default pg_hba.conf with our own, to allow password auth, since this is the
     ## only method metacat currently supports (as of June 2023)
-    ## TYPE       DATABASE      USER          ADDRESS         METHOD
-    pgHbaConfiguration: |
-      host        metacat       metacat       0.0.0.0/0       password
-      host        metacat       metacat       ::0/0           password
-      host        all           postgres      0.0.0.0/0       password
-      host        all           postgres      ::0/0           password
-      local       all           all                           trust
-
-    ## @param postgresql.primary.extendedConfiguration Extended configuration, appended to defaults
-    ## ref: https://github.com/bitnami/containers/tree/main/bitnami/postgresql#allow-settings-to-be-loaded-from-files-other-than-the-default-postgresqlconf
-    ## Must exceed metacat.database.maximumConnections, since some connections will be reserved for
-    ## superuser
+    ##  TYPE        DATABASE      USER          ADDRESS         METHOD
+    pg_hba:
+      - host        metacat       metacat       0.0.0.0/0       password
+      - host        metacat       metacat       ::0/0           password
+      - host        all           postgres      0.0.0.0/0       password
+      - host        all           postgres      ::0/0           password
+
+    ## @param cnpg.postgresql.pg_ident [default: see values.yaml] username mappings: pg_ident.conf
+    ## ref: https://cloudnative-pg.io/documentation/1.20/postgresql_conf/#the-pg_ident-section
+    ## https://www.postgresql.org/docs/current/auth-username-maps.html
+    ## Allows metacat DB user to log in without specifying a password or hostname, using:
+    ##   $ psql -U metacat
     ##
-    extendedConfiguration: |-
-      max_connections = 250
-
-    ## Postgresql persistence
-    ## For explanatory notes, see top-level `persistence` section for metacat, and the bitnami
-    ## postgresql README at https://github.com/bitnami/charts/blob/main/bitnami/postgresql/README.md
-    ## @param postgresql.primary.persistence.enabled Enable data persistence using PVC
-    ## @param postgresql.primary.persistence.existingClaim Existing PVC to re-use
-    ## @param postgresql.primary.persistence.storageClass Storage class of backing PV
-    ## @param postgresql.primary.persistence.size PVC Storage Request for postgres volume
+    pg_ident:
+      - local       postgres       metacat
```

Comment:
Does this map the postgres user to the metacat user, or vice versa? I'm not sure this mapping is a good idea because postgres is an admin account, but let's discuss how it works and what exposure this might cause..

Reply:
ok, that's fine. the `all all trust` model doesn't follow the principle of least privilege, so we should discuss that sometime, but if your new rule just does what we used to do, then that is fine for now. Given that the password is in the secret anyways, anyone that can exec into the container can get the pw anyway so it probably makes little difference.